src/test/java/org/fogbowcloud/manager/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -0,0 +1,382 @@
+		this.order = Mockito.mock(ComputeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    when(computePlugin.requestInstance(any(ComputeOrder.class), any(Token.class))).thenReturn(null);
+		this.localCloudConnector.requestInstance(order);
+	}
+	
+	
+	
+	// The order has an InstanceID, so the method getResourceInstance() is called.
+	@Test
+	public void testGetNetworkInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(NetworkOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.NETWORK);
+	    when(this.order.getInstanceId()).thenReturn(FAKE_INSTANCE_ID);
+	    when(networkPlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.networkInstance);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order has an InstanceID, so the method getResourceInstance() is called.
+	@Test
+	public void testGetVolumeInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(VolumeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.VOLUME);
+	    when(this.order.getInstanceId()).thenReturn(FAKE_INSTANCE_ID);
+	    when(volumePlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.volumeInstance);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order has an InstanceID, so the method getResourceInstance() is called.
+	@Test
+	public void testGetAttachmentInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(AttachmentOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.ATTACHMENT);
+	    when(this.order.getInstanceId()).thenReturn(FAKE_INSTANCE_ID);
+	    when(attachmentPlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.attachmentInstance);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order has an InstanceID, so the method getResourceInstance() is called.
+	// addReverseTunnelInfo() is called in this case.
+	@Test
+	public void testGetComputeInstance() throws FogbowManagerException, UnexpectedException {
+		
+		LocalCloudConnector localCloudConnectorSpy = Mockito.spy(this.localCloudConnector);
+		Mockito.doNothing().when(localCloudConnectorSpy).addReverseTunnelInfo(any(String.class), any(ComputeInstance.class));
+
+		this.order = Mockito.mock(ComputeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    when(this.order.getInstanceId()).thenReturn(FAKE_INSTANCE_ID);
+	    when(computePlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.computeInstance);
+		assertEquals(FAKE_INSTANCE_ID, localCloudConnectorSpy.getInstance(order).getId());
+	}
+	
+	
+	// The order doesn't have an InstanceID, so an empty NetworkInstance is returned with the same id of order.
```

Comment:
I'm no sure, maybe we need to check this Instances are "empty" @jvmafra talk to @fgan1 to discuss about it (I'm inclined to say it is ok, already)

Reply:
I had doubts in this method. The description says "When there is no instance, an empty one is created with the appropriate state", but the new instance is created only with id, without state. We can discuss on monday if this behavior is correct. 

About the test, we really can check if the instance is empty. All de attributes must be null, except id.